client/src/app/shared/forms/form-validators/user-validators.service.ts

```diff
@@ -23,15 +23,15 @@ export class UserValidatorsService {
     this.USER_USERNAME = {
       VALIDATORS: [
         Validators.required,
-        Validators.minLength(3),
-        Validators.maxLength(20),
-        Validators.pattern(/^[a-z0-9._]+$/)
+        Validators.minLength(1),
+        Validators.maxLength(50),
+        Validators.pattern(/^[a-zA-Z0-9]+[a-zA-Z0-9-._]+$/)
```

Comment:
Your regular expression is preventing `minLength` from being useful: the `pattern` test fails before the `minLength`.

Shouldn't the regex be `^[a-zA-Z0-9-._]+$` ?

Reply:
The reason I made it that way is to not allow users to have a username that starts with a `.`, `-`, `_`, so if they should want a single letter username it would have to be alphanumeric

Reply:
that regular expression is actually buggy, it allows minimum 2 chars, I had to fix it